src/vscode-dts/vscode.proposed.statusBarItemTooltip.d.ts

```diff
@@ -0,0 +1,13 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	// https://github.com/microsoft/vscode/issues/234339
+
+	export interface StatusBarItem {
+		tooltip2?: (token: CancellationToken) => ProviderResult<string | MarkdownString>;
```

Comment:
Can we not overload the existing `tooltip` method? Also, I would suggest to include the support for returning a 
```ts
tooltip: string | MarkdownString | ProviderResult<string | MarkdownString> | { readonly commands: Command[]; readonly contents: string | MarkdownString | ProviderResult<string | MarkdownString> }
```